### ISSUE TITLE
Corner conveyer belt does not work when getting shot at the same time.

### ISSUE THREAD

**Issue by marcelpanse** [marcelpanse/roborally#83]
The feedback from maska@cvx.fi on March 17th 2015 was:

> When I finish my movement on a conveyer belt that takes me to a conveyer belt corner square and take damage before the conveyer moves me to the corner, I go to the corner and immediately teleport back. This ruins my planning.

View it on [Doorbell](https://doorbell.io/applications/970/feedback/9851)

**Comment by marcelpanse** [comment]
"yes it is only when I take damage, but not only on corners. It does the 
same when entering straight belts as well. We didn't have express belts on 
the map so I don't know whether this happens on them as well. This 
teleporting back is very strange. It even put me and mine and my friend's 
bot on the exact same square. 

Thanks for your endeavor for this great game! 

–Masi "

**Comment by marcelpanse** [comment]
@xithan - can you write a test for this and see what happens? It is too hard for me to debug. It seems like the laser logic still uses the 'old' positions (pre-roller moves). I noticed when a player got moved one step by a roller, he still got shot on the old line of sight, then the checkRespawnsAndUpdateDb method also uses the positions and updates the player back to the square pre-roller move..

**Comment by xithan** [comment]
I tried it by hand and couldn't reproduce it but I had a look at the code and I think I found the reason. `checkRespawnAndUpdateDb` gets called with a timer. I guess the first robot that moves has a chance to get updated before the next phase starts but for the last robot it's likely that the update is executed too late so that it is still at its old position when the lasers start.  Fix is coming soon...
